ExeterTeam25/group-software-project
# Login register fix

To Test:
- Create a user using the registration form
- With an admin account check the new user is in the user group
- Do the same with SSO
- Change the username of the SSO account
- try creating a user with the same name as the SSO account (should display a message saying username is already taken)
- try creating a user with the same name but different capitalization, for example, CoNor instead of Conor, as the SSO account  (should display a message saying username is already taken)

Details:
- When a user logs in with SSO they are assigned a random username which can then be changed
- When a user registers their name is checked properly (for example, a user with the name conor can't create a user with the username cOnor)
- when a user registers with the registration form and SSO they are added to the 'user' group
- removed the ability to change your email as we don't want this to be possible
- In the register form the first name and last name has been added as a field
- re-formated code
